MetaMask/metamask-extension
# Ens multicoin

## Explanation

This PR is to resolve https://github.com/MetaMask/specifications/issues/11

## Screenshots/Screencaps

<img width="557" alt="Screenshot 2023-03-13 at 15 13 59" src="https://user-images.githubusercontent.com/2630/224744783-bfc5ef6c-1c19-4515-a33d-61c6b6d3c743.png">
<img width="572" alt="Screenshot 2023-03-13 at 15 13 45" src="https://user-images.githubusercontent.com/2630/224744790-4118f54b-e3e4-4629-9fbf-7d7d638ff417.png">
<img width="571" alt="Screenshot 2023-03-13 at 15 13 33" src="https://user-images.githubusercontent.com/2630/224744796-5ecbfc32-c57c-4956-b9e8-d64c46d171b6.png">


### Before

- When Metamask was connected to Mainnet, it resolves ETH address on ENS Mainnet
- When Metamask was connected to Testnet (Goerli, Ropsten, Rinkeby), it resolves ETH address on ENS Mainnet

### After

- When Metamask was connected to Mainnet, it resolves ETH address on ENS Mainnet
- When Metamask was connected to Testnet (Goerli, Ropsten, Rinkeby), it resolves ETH address on ENS Mainnet
- When Metamask was connected to Other network(eg: OP), it resolves `OP` address on Optimism network

## Manual Testing Steps

- When Metamask was connected to Mainnet, looking up `matoken.eth` should return 0x5a
- When Metamask was connected to Mainnet, looking up `makoto.eth` should return 0xfFD
- When Metamask was connected to Mainnet, looking up `0xmatoken.eth` should return `no address has been set for this name`

- When Metamask was connected to Testnet (Goerli), looking up `makoto.eth` should return `no address has been set for this name`
- When Metamask was connected to Testnet (Goerli), looking up `0xmatoken.eth` should return 0x5a

- When Metamask was connected to Other network(eg: OP), looking up `matoken.eth` should return `no address has been set for this name`
- When Metamask was connected to Other network(eg: OP), looking up `bertho.eth` should return `0x837`


## Pre-merge author checklist

  - [x ] How this problem was solved
  - [x ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added = The majority of test on ens.js can be moved into ethers.js once it's upgraded into v6 plugin (currently under discussion with metamask team and ethers.js team)

## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
